libraries/lib-components/EffectInterface.cpp

```diff
@@ -56,6 +56,42 @@ bool EffectDefinitionInterface::IsHiddenFromMenus()
    return false;
 }
 
+auto EffectDefinitionInterfaceEx::MakeSettings() const -> Settings
+{
+   // Temporary default implementation just saves self
+   // Cast away const! Capture pointer to self
+   return Settings( const_cast<EffectDefinitionInterfaceEx*>(this) );
```

Comment:
Settings object always refer to a current state, so `RealtimeEffectState` too. This is for another PR, right?

Reply:
I'm not sure I understand the question.

The Settings object has now way to hold a reference to a RealtimeEffectState object.  But, each RealtimeEffectState object will hold an instance of Settings.  Maybe you meant that.

Besides which, the global registry of effects, EffectsManager, will hold Settings.  That is for purposes of applying effects in the old, destuctive way.  Audacity now has a memory of the last settings you used for each effect, and applies the effect without prompting you again with a dialog, if you use the "Repeat Last Effect" command.  Also those settings appear again if you do reopen the dialog for another destructive application.

So if the Effect object becomes stateless, EffectManager must store one more Settings with the effect in the registry.